modules/renter/hostdb/hostdb.go

```diff
@@ -107,7 +107,9 @@ func newHostDB(g modules.Gateway, cs modules.ConsensusSet, persistDir string, de
 	hdb.hostTree = hosttree.New(hdb.calculateHostWeight)
 
 	// Load the prior persistence structures.
+	hdb.mu.Lock()
 	err = hdb.load()
+	hdb.mu.Unlock()
```

Comment:
It's an efficiency thing. The load function already touches all 1000 hosts. If we were to call queue scan elsewhere we'd need to touch all 1000 hosts again, or have the load function return a list of hosts that need to be queued